### ISSUE TITLE
Short circuited assignment in 'if' expression causes compilation error 

### ISSUE THREAD

**Issue by szabolcsmaj** [GoodGrind/ghostwriter#17]
If a conditional assignment is present in the 'if' expression, GW will generate code that leads to "might not have been initialized" error.

To reproduce:

    public void valueAssignmentInIfExpression() {
        int[] a = {1, 2, 3};
        int b, c;
        if ((b = a[0]) == 1 || (c = a[1]) == 3) {
            a = null;
        }
    }

In this case, adding the GW.valueChange calls to the 'if' statement body would also lead to the same compilation failure.
The solution for now is to ignore the if statement's conditional part when checking for value change events.

This is no longer a critical issue since the compilation error is fixed. The proper fix depends on #16, without that we cannot add proper value change events. On hold for now.
